appinventor/appengine/src/com/google/appinventor/shared/rpc/project/youngandroid/YoungAndroidProjectNode.java

```diff
@@ -100,5 +100,12 @@ public YoungAndroidComponentsFolder getComponentsFolder() {
     // Should never happen!
     throw new IllegalStateException("Couldn't find component folder");
   }
-
+  
+  @Override
+  public boolean hasExtensions(){
+    for (ProjectNode node : getComponentsFolder().getChildren()){
```

Comment:
So I took a look at this from our discussion on your previous PR, and it is safe for us to change the return type of the `getChildren()` method to `List<ProjectNode>` from `Iterable<ProjectNode>`. You could then use `.size()` here rather than having to iterate over the list, which will be more efficient.

Reply:
can we use the isEmpty() function?